src/main/java/Router.java

```diff
@@ -8,69 +8,62 @@
 
 public class Router implements UI {
 
-    private Board board = new Board();
-    private UI ui = this;
-    private Game game = new Game(ui, board);
-
+    private game.GridForWebConverter converter = new game.GridForWebConverter();
 
     public void run() {
-        game.playerSetUp();
-        get("/", (request, response) -> displayCurrentStateOfGame());
+        get("/", (request, response) -> displayCurrentStateOfGame(new Board()));
         get("/makeMove/:move", (request, response) -> makeMoveAndUpdateBoard(request));
     }
 
-    public String createQueryValueForGridState(List<Mark> grid) {
-        StringBuilder stringBuilder = new StringBuilder();
-        for (int markPosition = 0; markPosition < grid.size(); markPosition++) {
-            if (grid.get(markPosition).equals(Mark.EMPTY)) {
-                stringBuilder.append(String.format("%s", (markPosition + 1)));
-            } else {
-                stringBuilder.append(String.format("%s", grid.get(markPosition).toString()));
-            }
-        }
-        return stringBuilder.toString();
-    }
-
     private String makeMoveAndUpdateBoard(Request request) {
-        String move = request.params("move");
-        WebApplicationPlayer player = (WebApplicationPlayer) game.currentPlayer;
+        String move = request.queryParams("move");
+        String currentBoard = request.queryParams("currentBoard");
+        Board board = new Board(3, converter.convertToGridOfMarks(currentBoard));
+        Game newGame = new Game(this, board);
+        newGame.playerSetUp();
+        WebApplicationPlayer player = (WebApplicationPlayer) newGame.currentPlayer;
         player.receiveMove(move);
-        game.run();
-        return displayCurrentStateOfGame();
+        newGame.run();
+        return displayCurrentStateOfGame(board);
     }
 
-    private String displayCurrentStateOfGame() {
+    private String displayCurrentStateOfGame(Board board) {
         StringBuilder stringBuilder = new StringBuilder();
```

Comment:
Curious why you're using StringBuilders all over the place. It used to be that you'd use that class when performance suffered, but I'm pretty sure that the compiler takes care of that since like Java 7. Would this be more readable if you were to just use raw String concatenation (+)? Passing the stringBuilder around is a smell to me as well since you're mutating the thing and we want to minimise that.